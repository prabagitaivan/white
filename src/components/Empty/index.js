import React, { memo } from 'react'
import { makeStyles } from '@material-ui/styles'

const useStyles = makeStyles(
  {
    root: {
      textAlign: 'center',
      marginTop: '40vh'
    }
  },
  { name: 'Empty' }
)

export default memo(() => {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <div>Still Empty</div>
      <span role='img' aria-label='three-wise-monkeys'>
        ğŸ™ˆğŸ™‰ğŸ™Š
      </span>
    </div>
  )
})
