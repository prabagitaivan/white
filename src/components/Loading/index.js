import React, { memo } from 'react'
import { LinearProgress } from '@material-ui/core'
import { makeStyles } from '@material-ui/styles'

const useStyles = makeStyles(
  {
    root: {
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      marginTop: '40vh'
    },
    linearProgress: {
      height: 10,
      width: '60%',

      '& .MuiLinearProgress-barColorSecondary': {
        backgroundColor: '#b1b1b1'
      }
    }
  },
  { name: 'Loading' }
)

export default memo(() => {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <div>Loading</div>
      <LinearProgress color='secondary' className={classes.linearProgress} />
    </div>
  )
})
