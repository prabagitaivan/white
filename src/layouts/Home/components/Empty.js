import React, { PureComponent } from 'react'
import { empty as styles } from '../styles'

export default class Empty extends PureComponent {
  render () {
    return (
      <div style={styles.root}>
        <div>Still Empty</div>
        <span role='img' aria-label='three-wise-monkeys'>
          ðŸ™ˆðŸ™‰ðŸ™Š
        </span>
      </div>
    )
  }
}
