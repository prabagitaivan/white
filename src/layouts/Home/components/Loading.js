import React, { PureComponent } from 'react'
import { LinearProgress } from '@material-ui/core'
import { loading as styles } from '../styles'

export default class Loading extends PureComponent {
  render () {
    return (
      <div style={styles.root}>
        <div>Loading</div>
        <LinearProgress color='secondary' style={styles.linearProgress} />
      </div>
    )
  }
}
