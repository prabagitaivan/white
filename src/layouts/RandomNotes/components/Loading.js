import React, { memo } from 'react'
import { LinearProgress } from '@material-ui/core'
import { loading as useStyles } from '../styles'

export default memo(() => {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <div>Loading</div>
      <LinearProgress color='secondary' className={classes.linearProgress} />
    </div>
  )
})
