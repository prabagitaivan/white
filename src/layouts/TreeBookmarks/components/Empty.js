import React, { memo } from 'react'
import { empty as useStyles } from '../styles'

export default memo(() => {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <div>Still Empty</div>
      <span role='img' aria-label='three-wise-monkeys'>
        ☁️🌧️⛈️
      </span>
    </div>
  )
})
