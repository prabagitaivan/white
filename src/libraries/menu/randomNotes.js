import { FlashOn, Loop } from '@material-ui/icons'
import { shuffle, random } from 'lodash'
import store from '../../stores'
import { edit } from '../../reducers/randomNotes'

const options = {
  left: [
    {
      Icon: Loop,
      text: 'shuffle-notes',
      action: () => {
        const { data } = store.getState().randomNotes
        if (data && data.length > 0) {
          store.dispatch(edit(shuffle(data)))
        }
      }
    }
  ],
  right: [
    {
      Icon: FlashOn,
      text: 'open-random',
      action: () => {
        const { data } = store.getState().randomNotes
        if (data && data.length > 0) {
          window.open(data[random(data.length - 1)].url)
        }
      }
    }
  ]
}

export default options
