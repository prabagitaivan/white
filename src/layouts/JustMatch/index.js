import React, { memo, useEffect, useState } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { Grid, IconButton, InputBase } from '@mui/material'
import { Search } from '@mui/icons-material'
import { makeStyles } from '@mui/styles'
import { shuffle } from 'lodash'
import Navigator from '../../components/Navigator'
import Content from '../../components/Content'
import { setPage } from '../../reducers/status'
import Loading from '../../components/Loading'
import Empty from '../../components/Empty'
import useData from './hooks/useData'

const useStyles = makeStyles(
  {
    search: {
      display: 'flex',
      alignItems: 'center',
      width: '100%',
      padding: '2px 4px 2px 18px',
      marginTop: 20,
      marginBottom: 25,
      borderRadius: 100,
      backgroundColor: 'transparent',
      border: '1px solid #b1b1b1'
    },
    input: {
      flexGrow: 1,
      fontSize: 12
    },
    icon: {
      padding: ({ desktop }) => (desktop ? 2 : 4)
    },
    link: {
      cursor: 'pointer',
      textDecoration: 'none',
      width: 'fit-content',
      color: ({ light }) => (light ? '#181818' : '#ffffff'),
      '& > div:first-child': {
        fontSize: 10,
        fontWeight: 'lighter'
      },
      '& > div:last-child': {
        fontSize: 18,
        fontWeight: 'bolder'
      },

      '&:hover': {
        '& > div:last-child': {
          textDecoration: 'underline'
        }
      }
    },
    description: {
      fontSize: 12
    }
  },
  { name: 'JustMatch' }
)

export default memo(() => {
  const { desktop, light } = useSelector(state => state.status)
  const { data, requesting } = useData()
  const [keyword, setKeyword] = useState('')
  const dispatch = useDispatch()
  const classes = useStyles({ desktop, light })

  useEffect(() => {
    dispatch(setPage('JustMatch'))
  }, [dispatch])

  const onChange = event => setKeyword(event.target.value)
  const openMatch = url => window.open(url)

  return (
    <div>
      <Navigator />
      <Content>
        {requesting ? (
          <Loading />
        ) : data.length === 0 ? (
          <Empty />
        ) : (
          <div>
            <div className={classes.search}>
              <InputBase
                value={keyword}
                onChange={onChange}
                className={classes.input}
                placeholder='match'
                autoFocus
              />
              <IconButton size='small' disabled>
                <Search className={classes.icon} />
              </IconButton>
            </div>
            <Grid container spacing={3}>
              {shuffle(data)
                .filter(
                  item =>
                    keyword === '' ||
                    item.title.toLowerCase().includes(keyword) ||
                    item.description.toLowerCase().includes(keyword)
                )
                .map((item, index) => (
                  <Grid item key={index} xs={12}>
                    <div
                      onClick={() => openMatch(item.url)}
                      className={classes.link}
                    >
                      <div className={classes.url}>{item.url}</div>
                      <div className={classes.title}>{item.title}</div>
                    </div>
                    <div className={classes.description}>
                      {item.description}
                    </div>
                  </Grid>
                ))}
            </Grid>
          </div>
        )}
      </Content>
    </div>
  )
})
