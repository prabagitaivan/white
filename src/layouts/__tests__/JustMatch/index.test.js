import { render, fireEvent } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'
import { Provider } from 'react-redux'
import { createStore } from '../../../stores'
import Styles from '../../../styles'
import { request as requestNotes } from '../../../reducers/randomNotes'
import { request as requestBookmarks } from '../../../reducers/treeBookmarks'
import { setPage } from '../../../reducers/status'
import Loading from '../../../components/Loading'
import Empty from '../../../components/Empty'
import JustMatch from '../../JustMatch'

let store

function renderJustMatch ({
  desktop = true,
  light = true,
  randomNotes,
  treeBookmarks
}) {
  const preloadedState = {
    status: { desktop, light, page: 'JustMatch' },
    randomNotes,
    treeBookmarks
  }
  store = createStore(preloadedState)

  jest.spyOn(store, 'dispatch').mockReturnValue()

  return render(
    <Provider store={store}>
      <Styles>
        <JustMatch />
      </Styles>
    </Provider>
  )
}

afterEach(() => {
  store.dispatch.mockRestore()
})

describe('layouts RandomNotes', () => {
  describe('snapshots', () => {
    it('contain navigator', () => {
      const { container } = renderJustMatch({})
      const Navigator = container.querySelector('#navigator')
      expect(Navigator).toBeInTheDocument()
    })
    it('contain loading content when requesting', () => {
      const { container } = renderJustMatch({})
      const LoadingRender = render(
        <Provider store={store}>
          <Styles>
            <Loading />
          </Styles>
        </Provider>
      )
      const Content = container.querySelector('#content')
      const LoadingContent = LoadingRender.container
      expect(Content.firstChild).toEqual(LoadingContent.firstChild)
    })
    it('contain empty content when requesting result is empty', () => {
      const { container } = renderJustMatch({
        randomNotes: { requesting: false, data: [] },
        treeBookmarks: { requesting: false, data: [] }
      })
      const EmptyRender = render(
        <Provider store={store}>
          <Styles>
            <Empty />
          </Styles>
        </Provider>
      )
      const Content = container.querySelector('#content')
      const EmptyContent = EmptyRender.container
      expect(Content.firstChild).toEqual(EmptyContent.firstChild)
    })
    it('contain data content when requesting result is not empty in desktop', () => {
      const { container } = renderJustMatch({
        randomNotes: {
          requesting: false,
          data: [
            { image: 'image1', title: 'title1', url: 'url1', active: true }
          ]
        },
        treeBookmarks: {
          requesting: false,
          data: { subdata2: [{ title: 'title2', url: 'url2', active: true }] }
        }
      })
      const Content = container.querySelector('#content')

      expect(Content).toBeInTheDocument()
      expect(Content.firstChild).toMatchSnapshot()
    })
    it('contain data content when requesting result is not empty in mobile', () => {
      const { container } = renderJustMatch({
        desktop: false,
        randomNotes: {
          requesting: false,
          data: [
            { image: 'image1', title: 'title1', url: 'url1', active: true }
          ]
        },
        treeBookmarks: {
          requesting: false,
          data: { subdata2: [{ title: 'title2', url: 'url2', active: true }] }
        }
      })
      const Content = container.querySelector('#content')

      expect(Content).toBeInTheDocument()
      expect(Content.firstChild).toMatchSnapshot()
    })
    it('contain data content when requesting result is not empty in dark mode', () => {
      const { container } = renderJustMatch({
        light: false,
        randomNotes: {
          requesting: false,
          data: [
            { image: 'image1', title: 'title1', url: 'url1', active: true }
          ]
        },
        treeBookmarks: {
          requesting: false,
          data: { subdata2: [{ title: 'title2', url: 'url2', active: true }] }
        }
      })
      const Content = container.querySelector('#content')

      expect(Content).toBeInTheDocument()
      expect(Content.firstChild).toMatchSnapshot()
    })
    it('show active only in data content', () => {
      const { queryAllByText, queryByText } = renderJustMatch({
        randomNotes: {
          requesting: false,
          data: [
            { image: 'image1', title: 'title1', url: 'url1', active: false },
            { image: 'image2', title: 'title2', url: 'url2', active: true }
          ]
        },
        treeBookmarks: {
          requesting: false,
          data: { subdata3: [{ title: 'title3', url: 'url3', active: false }] }
        }
      })

      const notes = queryAllByText('random notes')
      const bookmarks = queryAllByText('tree bookmarks', { exact: false })
      expect(notes.length).toEqual(2) // navigator and 1 match item
      expect(bookmarks.length).toEqual(1) // navigator only

      expect(queryByText('title1')).not.toBeInTheDocument()
      expect(queryByText('title2')).toBeInTheDocument()
    })
    it('show existed only in data content', () => {
      const { queryAllByText, queryByText } = renderJustMatch({
        randomNotes: {
          requesting: false,
          data: []
        },
        treeBookmarks: {
          requesting: false,
          data: { subdata: [{ title: 'title', url: 'url', active: true }] }
        }
      })

      const notes = queryAllByText('random notes')
      const bookmarks = queryAllByText('tree bookmarks', { exact: false })
      expect(notes.length).toEqual(1) // navigator only
      expect(bookmarks.length).toEqual(2) // navigator and 1 match item

      expect(queryByText('title')).toBeInTheDocument()
    })
  })
  describe('mounting', () => {
    it('set page to random notes after mount', () => {
      renderJustMatch({})
      expect(store.dispatch).toHaveBeenCalledWith(setPage('JustMatch'))
    })
    it('request for random notes and tree bookmarks data after mount', () => {
      renderJustMatch({})
      expect(store.dispatch).toHaveBeenCalledWith(requestNotes())
      expect(store.dispatch).toHaveBeenCalledWith(requestBookmarks())
    })
    it('not request for random notes if data exists', () => {
      renderJustMatch({
        randomNotes: {
          requesting: false,
          data: [{ image: 'image', title: 'title', url: 'url', active: true }]
        }
      })
      expect(store.dispatch).not.toHaveBeenCalledWith(requestNotes())
    })
    it('not request for tree bookmarks if data exists', () => {
      renderJustMatch({
        treeBookmarks: {
          requesting: false,
          data: { subdata: [{ title: 'title', url: 'url', active: true }] }
        }
      })
      expect(store.dispatch).not.toHaveBeenCalledWith(requestBookmarks())
    })
  })
  describe('userEvent', () => {
    beforeEach(() => {
      jest.spyOn(window, 'open').mockReturnValue()
    })
    afterEach(() => {
      window.open.mockRestore()
    })

    it("open url when click the match's item", () => {
      const { getByText } = renderJustMatch({
        randomNotes: {
          requesting: false,
          data: [
            { image: 'image1', title: 'title1', url: 'url1', active: true }
          ]
        },
        treeBookmarks: {
          requesting: false,
          data: { subdata2: [{ title: 'title2', url: 'url2', active: true }] }
        }
      })

      const Item1 = getByText('title1')
      userEvent.click(Item1)
      expect(window.open).toHaveBeenCalledWith('url1')

      const Item2 = getByText('url2')
      userEvent.click(Item2)
      expect(window.open).toHaveBeenCalledWith('url2')
    })

    it('match input with the items', async () => {
      const { container, findAllByRole } = renderJustMatch({
        randomNotes: {
          requesting: false,
          data: [
            { image: 'image1', title: 'title1', url: 'url1', active: true },
            { image: 'image2', title: 'title2', url: 'url2', active: true }
          ]
        },
        treeBookmarks: {
          requesting: false,
          data: { subdata3: [{ title: 'title3', url: 'url3', active: true }] }
        }
      })

      const MatchTextBoxes = await findAllByRole('textbox')

      fireEvent.change(MatchTextBoxes[0], { target: { value: 'notes' } })
      const Items = container.querySelector('#content').firstChild.lastChild
      expect(Items).toMatchSnapshot()

      fireEvent.change(MatchTextBoxes[0], { target: { value: 'title3' } })
      expect(Items).toMatchSnapshot()

      fireEvent.change(MatchTextBoxes[0], { target: { value: 'asdf' } })
      expect(Items).toMatchSnapshot()
    })
  })
})
