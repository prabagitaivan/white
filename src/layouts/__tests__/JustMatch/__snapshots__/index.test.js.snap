// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layouts RandomNotes snapshots contain data content when requesting result is not empty in dark mode 1`] = `
<div>
  <div
    class="WhiteJustMatch-search"
  >
    <div
      class="MuiInputBase-root MuiInputBase-colorPrimary Mui-focused WhiteJustMatch-input css-utwxka-MuiInputBase-root"
    >
      <input
        class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
        placeholder="match"
        type="text"
        value=""
      />
    </div>
    <button
      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium WhiteJustMatch-icon WhiteJustMatch-icon css-i4bv87-MuiSvgIcon-root"
        data-testid="SearchIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </button>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
    >
      <div
        class="WhiteJustMatch-link WhiteJustMatch-link"
      >
        <div>
          url1
        </div>
        <div>
          title1
        </div>
      </div>
      <div
        class="WhiteJustMatch-description"
      >
        random notes
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
    >
      <div
        class="WhiteJustMatch-link WhiteJustMatch-link"
      >
        <div>
          url2
        </div>
        <div>
          title2
        </div>
      </div>
      <div
        class="WhiteJustMatch-description"
      >
        tree bookmarks - subdata2
      </div>
    </div>
  </div>
</div>
`;

exports[`layouts RandomNotes snapshots contain data content when requesting result is not empty in desktop 1`] = `
<div>
  <div
    class="WhiteJustMatch-search"
  >
    <div
      class="MuiInputBase-root MuiInputBase-colorPrimary Mui-focused WhiteJustMatch-input css-i2fblb-MuiInputBase-root"
    >
      <input
        class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
        placeholder="match"
        type="text"
        value=""
      />
    </div>
    <button
      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium WhiteJustMatch-icon WhiteJustMatch-icon css-i4bv87-MuiSvgIcon-root"
        data-testid="SearchIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </button>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
    >
      <div
        class="WhiteJustMatch-link WhiteJustMatch-link"
      >
        <div>
          url1
        </div>
        <div>
          title1
        </div>
      </div>
      <div
        class="WhiteJustMatch-description"
      >
        random notes
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
    >
      <div
        class="WhiteJustMatch-link WhiteJustMatch-link"
      >
        <div>
          url2
        </div>
        <div>
          title2
        </div>
      </div>
      <div
        class="WhiteJustMatch-description"
      >
        tree bookmarks - subdata2
      </div>
    </div>
  </div>
</div>
`;

exports[`layouts RandomNotes snapshots contain data content when requesting result is not empty in mobile 1`] = `
<div>
  <div
    class="WhiteJustMatch-search"
  >
    <div
      class="MuiInputBase-root MuiInputBase-colorPrimary Mui-focused WhiteJustMatch-input css-i2fblb-MuiInputBase-root"
    >
      <input
        class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
        placeholder="match"
        type="text"
        value=""
      />
    </div>
    <button
      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium WhiteJustMatch-icon WhiteJustMatch-icon css-i4bv87-MuiSvgIcon-root"
        data-testid="SearchIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </button>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
    >
      <div
        class="WhiteJustMatch-link WhiteJustMatch-link"
      >
        <div>
          url1
        </div>
        <div>
          title1
        </div>
      </div>
      <div
        class="WhiteJustMatch-description"
      >
        random notes
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
    >
      <div
        class="WhiteJustMatch-link WhiteJustMatch-link"
      >
        <div>
          url2
        </div>
        <div>
          title2
        </div>
      </div>
      <div
        class="WhiteJustMatch-description"
      >
        tree bookmarks - subdata2
      </div>
    </div>
  </div>
</div>
`;

exports[`layouts RandomNotes userEvent match input with the items 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
  >
    <div
      class="WhiteJustMatch-link WhiteJustMatch-link"
    >
      <div>
        url1
      </div>
      <div>
        title1
      </div>
    </div>
    <div
      class="WhiteJustMatch-description"
    >
      random notes
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
  >
    <div
      class="WhiteJustMatch-link WhiteJustMatch-link"
    >
      <div>
        url2
      </div>
      <div>
        title2
      </div>
    </div>
    <div
      class="WhiteJustMatch-description"
    >
      random notes
    </div>
  </div>
</div>
`;

exports[`layouts RandomNotes userEvent match input with the items 2`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
  >
    <div
      class="WhiteJustMatch-link WhiteJustMatch-link"
    >
      <div>
        url3
      </div>
      <div>
        title3
      </div>
    </div>
    <div
      class="WhiteJustMatch-description"
    >
      tree bookmarks - subdata3
    </div>
  </div>
</div>
`;

exports[`layouts RandomNotes userEvent match input with the items 3`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
/>
`;
